packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -800,6 +801,20 @@ export class Project extends ProjectBase {
               account: parsedArn?.account,
               region: parsedArn?.region,
             }));
+            // if secret comes from another account, SecretsManager will need to access
+            // KMS on the other account as well to be able to get the secret
+            if (parsedArn?.account !== stack.account) {
```

Comment:
I'm 99% sure this condition is incorrect 🙂. If `parsedArn` is `undefined`, we should never go into this `if` (because in that case we can't know whether the Secret is from a different account), but we do today (because `stack.account` is never `undefined`).

There's also a subtlety with how we need to compare the accounts, because they can be unspecified, so roughly equivalent to the `AWS::Region` intrinsic.

In summary, this line needs to become:

```suggestion
            if (parsedArn && Token.compareStrings(parsedArn.account, stack.account) === TokenComparison.DIFFERENT) {
```